docs_src/design/adr/system-management/0006-Metrics-Collection.md

```diff
@@ -0,0 +1,382 @@
+        - *Also, should there be an interval per metric or a single interval for all metrics collected and pushed?*
+            - Recommendation by @cloudxxx8 to support interval per metric but to have one for all as default
+
+!!! Info
+    Initially, it was proposed that metrics be associated with a "level" and allow metrics to be turned on or off by level (like levels associated to log messages in logging).  The level of metrics data seems arbitrary at this time and considered too complex for initial implementation.  This may be reconsidered in a future release and based on new requirements/use cases.
+
+    It was also proposed to categorize or label metrics - essentially allowing grouping of various metrics.  This would allow groups of metrics to be turned on or off, and allow metrics to be organized per the group when reporting.  At this time, this feature is also considered beyond the scope of the initial implementation and to be reconsidered in a future release based on requirements/use case needs.
+
+### Requested Metrics
+
+The following is a list of example metrics requested by the EdgeX community and adopters for various service areas.  Again, metrics would generally be collected and pushed to the message topic in some configured interval (example: 1/5/15 minutes or other defined interval).  The exact metrics collected by each service will be determined by the service implementers (or SDK implementers in the case of the app functions and device service SDKs).
+
+#### General
+
+The following metrics apply to all (or most) services.
+
+- Service uptime (time since last service boot)
+- Cumulative number of API requests succeeded / failed / invalid (2xx vs 5xx vs 4xx)
+- Avg response time (in milliseconds or appropriate unit of measure) on APIs
+- Avg and Max request size
+
+#### Core/Supporting
+
+- Latency (measure of time) an event takes to get through core data
+- Latency (measure of time) a command request takes to get to a device service
+- Indication of health – that events are being processed during a configurable period
+- Number of events in persistence
+- Number of readings in persistence
+- Number of validation failures (validation of device identification)
+- Number of notification transactions
+- Number of notifications handled
+- Number of failed notification transmissions
+- Number of notifications in retry status
+
+#### Application Services
+
+- Processing time for a pipeline; latency (measure of time) an event takes to get through an  application service pipeline
+- DB access times
+- How often are we failing export to be sent to db to be retried at a later time
+- What is the current store and forward queue size
+- How much data (size in KBs or MBs) of packaged sensor data is being sent to an endpoint (or volume)
+- Number of invalid messages that triggered pipeline
+- Number of events processed
+
+#### Device Services
+
+- Number of devices managed by this DS

```

Comment:
This is already possible to get from Core Metadata, and it also doesn't align with the type of metrics described earlier in this ADR.